### PR TITLE
Fix issue#383

### LINKED ISSUE #383: Incorrect project structure gives an unhelpful error message
Please answer these questions before submitting a bug report.

### What version of godog are you using?
```
0.11.0
```

### What version of Go are you using?
```
1.16.2
```

### What did you do?
Created a new project with the following structure,
```
|-- features
|   `-- health.feature
|-- main.go
|-- Dockerfile
|-- go.mod
`-- go.sum
```
My `main.go`,
```go
package main

import "github.com/cucumber/godog"

func InitializeScenario(ctx *godog.ScenarioContext) {

}
```
I wanted to confirm that my `Dockerfile` was correctly working and running this minimal example with `godog`. But it failed, I then tried running it with my local copy of `godog`, but that failed as well.

After doing the following change it worked,
```shell
$ mv main.go main_test.go
```

### What did you expect to see?
A more informative error message on why `godog` failed with my minimal example.

### What did you see instead?
```shell
$ godog
failed to compile testmain package: exit status 1 - output: compile: -importcfg: open /tmp/go-build1250231118/b001/importcfg.link: no such file or directory
```

### PR DESCRIPTION
**Pull request by ALCooper12**
### 🤔 What's changed?

- Added a check for missing test files and raised a more helpful error 
 

### ⚡️ What's your motivation? 

- trying to fix #383 

### 🏷️ What kind of change is this?

- :bug: Bug fix (non-breaking change which fixes a defect)

### 📋 Checklist:

- [x] I agree to respect and uphold the [Cucumber Community Code of Conduct](https://cucumber.io/conduct/)
- [x] I've changed the behaviour of the code
  - [x] I have added/updated tests to cover my changes.
- [x] Users should know about my change
  - [ ] I have added an entry to the "Unreleased" section of the [**CHANGELOG**](../CHANGELOG.md), linking to this pull request.

----

*This text was originally generated from a [template](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/about-issue-and-pull-request-templates), then edited by hand. [You can modify the template here.](https://github.com/cucumber/.github/edit/main/.github/PULL_REQUEST_TEMPLATE.md)*

### CODE REVIEW

**Comment by mattwynne** [comment]
@vearutop I am a bit stuck on how to advise @ALCooper12 to actually fix this. We've got it trapped with a failing test around `Builder.Build` as you suggested in Slack a few weeks ago.

It appears that the unexpected error we're seeing is because the `compilerCfg` file is missing from the build directory. The code beneath [here](https://github.com/cucumber/godog/blob/main/internal/builder/builder.go#L184) assumes that there will be a file at `compilerCfg` but in fact that file is only copied if we go into the `if` statement [here](https://github.com/cucumber/godog/blob/main/internal/builder/builder.go#L185).

I don't really understand how we end up in this situation just because the user has called their file `main.go` instead of `main_test.go`!?

Can you shed any light on it?

**Comment by ALCooper12** [comment]
I still need to update the change log, but otherwise this is ready! 
Please be gentle with feedback lolz

**Comment by mattwynne** [comment]
@vearutop any feedback?

File: internal/builder/builder_test.go [hunk 1]
```diff
@@ -237,6 +245,38 @@ func testWithoutTestSourceNotInGoPath(t *testing.T) {
...
+	if build.Default.GOOS == "windows" {
+		testBin += ".exe"
+	}
+
+	// call the builder - we should get an error
+	err = builder.Build(testBin)
+	// check that we even got an error at all
+	require.NotNil(t, err)
+	// now check the details of the error message
+	require.Contains(t, err.Error(), "incorrect project structure")
```
Comment (vearutop) [inline internal/builder/builder_test.go:277]:
```suggestion
	require.EqualError(t, err, "incorrect project structure: no test files found")
```
Let's make error assertion more detailed.

**Review by vearutop — APPROVED** [review]
I have a minor suggestion, overall this looks good to me! 👍

**Comment by aslakhellesoy** [comment]
Hi @ALCooper12,

Thanks for your making your first contribution to Cucumber, and welcome to the Cucumber committers team! You can now push directly to this repo and all other repos under the cucumber organization! 🍾

In return for this generous offer we hope you will:

* ✅ Continue to use branches and pull requests. When someone on the core team approves a pull request (yours or someone else's), you're welcome to merge it yourself.
* 💚 Commit to setting a good example by following and upholding our [code of conduct](https://cucumber.io/conduct) in your interactions with other collaborators and users.
* 💬 Join the [community Slack channel](http://cucumber.io/support#slack) to meet the rest of the team and make yourself at home.
* ℹ️ Don't feel obliged to help, just do what you can if you have the time and the energy.
* 🙋 Ask if you need anything. We're looking for feedback about how to make the project more welcoming, so please tell us!

On behalf of the Cucumber core team,
Aslak Hellesøy
Creator of Cucumber

**Comment by mattwynne** [comment]
🎉 congrats @ALCooper12 and thanks for all your hard work and patience figuring out this fix together. I've learned a lot about Go and godog's codebase working with you, I hope we can continue working together!

**Pull request merged**